docs/how_to/add_op_in_backend.md

```diff
@@ -0,0 +1,554 @@
+    return std::vector<std::string>{"data"};                                       // 10
+  })                                                                               // 11
+.set_attr<nnvm::FInferShape>("FInferShape", QuadraticOpShape)                      // 12
+.set_attr<nnvm::FInferType>("FInferType", QuadraticOpType)                         // 13
+.set_attr<FCompute>("FCompute<cpu>", QuadraticOpForward<cpu>)                      // 14
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_quadratic"})  // 15
+.set_attr<nnvm::FInplaceOption>("FInplaceOption",                                  // 16
+  [](const NodeAttrs& attrs) {                                                     // 17
+    return std::vector<std::pair<int, int> >{{0, 0}};                              // 18
+  })                                                                               // 19
+.add_argument("data", "NDArray-or-Symbol", "Input ndarray")                        // 20
+.add_arguments(QuadraticParam::__FIELDS__());                                      // 21
+
+NNVM_REGISTER_OP(_backward_quadratic)                                              // 22
+.set_attr_parser(ParamParser<QuadraticParam>)                                      // 23
+.set_num_inputs(2)                                                                 // 24
+.set_num_outputs(1)                                                                // 25
+.set_attr<nnvm::TIsBackward>("TIsBackward", true)                                  // 26
+.set_attr<FCompute>("FCompute<cpu>", QuadraticOpBackward<cpu>);                    // 27
+```
+
+- Line 1: Register the parameter struct.
+- Line 2: Register an operator named `quadratic` by creating an instance
+of `Op` type and save it in the operator manager and return a reference
+of the just created operator object.
+- Lines 3-4: Add description as an operator attribute
+including examples of the operator. The documentation engine would extract
+this description and display it on the documentation web page.
+- Line 5: Set parameter struct parser for the operator. It is used for parsing
+the parameters `a`, `b`, and `c` input from frontend.
+- Line 6: Set the number of inputs for the operator.
+- Line 7: Set the number of outputs for the operator.
+- Lines 8-11: Defines a function generating a vector of names of
+the operator input arguments. This function is used to add missing
+arguments that users did not specify when creating a symbolic operator.
+For example, `quad_func=mx.sym.quadratic()` is still a valid symbol
+since we have added the attribute `FListInputNames` to the operator node
+in the computational graph. MXNet would
+add the missing argument with name `quadratic0_data`, where the prefix
+`quadratic0` is the operator name appended with an index and the postfix
+`data` comes from the return value of the user defined `FListInputName` function.
```

Comment:
In the case of mx.sym.quadratic(), I understand that the computation graph can identify that it takes a variable, and creates a node in the graph with quadratic0_data. But such a function can't actually run because there's no argument, right? Can that variable be assigned somewhere later?

Reply:
The argument is assigned values during symbol binding stage. That could be another long tutorial. I will add a few sentences to make it clear here.